contract/src/launchIt.js

```diff
@@ -0,0 +1,227 @@
+// @ts-check
+import { Far } from '@endo/far';
+import { M, mustMatch } from '@endo/patterns';
+import { BrandShape, DisplayInfoShape } from '@agoric/ertp/src/typeGuards.js';
+import { AmountMath, AssetKind } from '@agoric/ertp/src/amountMath.js';
+import {
+  TimerServiceShape,
+  TimestampRecordShape,
+} from '@agoric/time/src/typeGuards.js';
+import { atomicRearrange } from '@agoric/zoe/src/contractSupport/index.js';
+import {
+  floorMultiplyBy,
+  makeRatio,
+} from '@agoric/zoe/src/contractSupport/ratio';
+
+const { Fail, quote: q } = assert;
+
+/** @type {import('./types').ContractMeta} */
+export const meta = {
+  customTermsShape: M.splitRecord(
+    { name: M.string(), supplyQty: M.bigint() },
+    { displayInfo: DisplayInfoShape },
+  ),
+};
+export const { customTermsShape } = meta;
+/**
+ * @typedef {{
+ *   name: string,
+ *   supplyQty: bigint,
+ *   displayInfo?: DisplayInfo,
+ * }} LaunchTerms
+ */
+
+const kw = s => (/^[A-Z]/.test(s) ? s : `KW${s}`); // only addresses initial cap
+
+/**
+ * @param {ZCF} zcf
+ * @param {string} kw
+ */
+const makeZcfIssuerKit = async (zcf, kw) => {
+  const mint = await zcf.makeZCFMint(kw);
+  const kit = mint.getIssuerRecord();
+  return { ...kit, mint };
+};
+
+/**
+ * This contract is limited to fungible assets.
+ *
+ * @param {ZCF<LaunchTerms>} zcf
+ * @param {unknown} _privateArgs
+ * @param {import('@agoric/vat-data').Baggage} _baggage
+ */
+export const start = async (zcf, _privateArgs, _baggage) => {
+  const { name, supplyQty, displayInfo = {}, brands } = zcf.getTerms();
+  mustMatch(brands, M.splitRecord({ Deposit: BrandShape }));
+  const { zcfSeat: deposits } = zcf.makeEmptySeatKit();
+
+  const assetMint = await zcf.makeZCFMint(kw(name), AssetKind.NAT, displayInfo);
+  const asset = assetMint.getIssuerRecord();
+  const share = await makeZcfIssuerKit(zcf, 'Share');
+
+  const ShapeAmt = {
+    Asset: harden({ brand: asset.brand, value: M.nat() }),
+    Deposit: harden({ brand: brands.Deposit, value: M.nat() }),
+    Share: harden({ brand: share.brand, value: M.nat() }),
+  };
+  const ExitDeadlineShape = {
+    afterDeadline: {
+      timer: TimerServiceShape,
+      deadline: TimestampRecordShape,
+    },
+  };
+  // export these from a client interface module?
+  const Shape = harden({
+    Amount: ShapeAmt,
+    Proposal: {
+      Mint: M.splitRecord({
+        exit: ExitDeadlineShape,
+      }),
+      Collect: M.splitRecord(
+        { exit: ExitDeadlineShape },
+        { want: { Deposit: ShapeAmt.Deposit } },
+      ),
+      Deposit: M.splitRecord(
+        { give: { Deposit: ShapeAmt.Deposit } },
+        { want: { Shares: ShapeAmt.Share } },
+      ),
+      Withdraw: M.splitRecord({
+        want: { Deposit: ShapeAmt.Deposit },
```

Comment:
This has different bracing than the previous lines. That seems more like a bug than a difference.